api/v1alpha1/types_test.go

```diff
@@ -0,0 +1,301 @@
+			wantValid: true,
+		},
+		"valid single character": {
+			name:      "a",
+			wantValid: true,
+		},
+		"valid alphanumeric": {
+			name:      "secret123",
+			wantValid: true,
+		},
+		"valid with hyphens": {
+			name:      "my-secret-123",
+			wantValid: true,
+		},
+		"invalid uppercase": {
+			name:      "My-Secret",
+			wantValid: false,
+		},
+		"invalid leading hyphen": {
+			name:      "-secret",
+			wantValid: false,
+		},
+		"invalid trailing hyphen": {
+			name:      "secret-",
+			wantValid: false,
+		},
+		"invalid underscore": {
+			name:      "my_secret",
+			wantValid: false,
+		},
+		"invalid dot": {
+			name:      "my.secret",
+			wantValid: false,
+		},
+		"empty name": {
+			name:      "",
+			wantValid: false,
+		},
+	}
+
+	for testName, tc := range tests {
+		t.Run(testName, func(t *testing.T) {
+			ref := SecretReference{Name: tc.name}
+
+			// Test that the struct can be created
+			assert.Equal(t, tc.name, ref.Name)
+
+			// The actual validation happens at the kubebuilder level,
+			// but we can test the pattern would match/not match
+			// Pattern: ^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+			if tc.wantValid {
+				assert.Regexp(t, `^[a-z0-9]([-a-z0-9]*[a-z0-9])?$`, ref.Name,
+					"Expected name %q to match validation pattern", ref.Name)
+			} else if tc.name != "" { // empty string is handled by Required validation
+				assert.NotRegexp(t, `^[a-z0-9]([-a-z0-9]*[a-z0-9])?$`, ref.Name,
+					"Expected name %q to NOT match validation pattern", ref.Name)
+			}
+		})
+	}
+}
+
+func TestTemporalConnectionReference_Validation(t *testing.T) {
+	tests := map[string]struct {
+		name      string
+		wantValid bool
+	}{
+		"valid lowercase name": {
+			name:      "my-temporal-connection",
+			wantValid: true,
+		},
+		"valid single character": {
+			name:      "t",
+			wantValid: true,
+		},
+		"valid alphanumeric": {
+			name:      "temporal123",
+			wantValid: true,
+		},
+		"valid with hyphens": {
+			name:      "temporal-conn-123",
+			wantValid: true,
+		},
+		"invalid uppercase": {
+			name:      "My-Connection",
+			wantValid: false,
+		},
+		"invalid leading hyphen": {
+			name:      "-connection",
+			wantValid: false,
+		},
+		"invalid trailing hyphen": {
+			name:      "connection-",
+			wantValid: false,
+		},
+		"invalid underscore": {
+			name:      "my_connection",
+			wantValid: false,
+		},
+		"invalid dot": {
+			name:      "my.connection",
+			wantValid: false,
+		},
+		"empty name": {
+			name:      "",
+			wantValid: false,
+		},
+	}
+
+	for testName, tc := range tests {
+		t.Run(testName, func(t *testing.T) {
+			ref := TemporalConnectionReference{Name: tc.name}
+
+			// Test that the struct can be created
+			assert.Equal(t, tc.name, ref.Name)
+
+			// The actual validation happens at the kubebuilder level,
+			// but we can test the pattern would match/not match
+			// Pattern: ^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+			if tc.wantValid {
+				assert.Regexp(t, `^[a-z0-9]([-a-z0-9]*[a-z0-9])?$`, ref.Name,
```

Comment:
Test the production validation logic instead of duplicating it here, or don't test it at all.